AppCenter/AppCenter/Internals/Storage/MSDBStorage.m

```diff
@@ -78,77 +127,78 @@ - (NSUInteger)countEntriesForTable:(NSString *)tableName condition:(nullable NSS
 }
 
 - (BOOL)executeNonSelectionQuery:(NSString *)query {
-  sqlite3 *db = NULL;
-  int result = SQLITE_OK;
-  result = sqlite3_open_v2([[self.dbFileURL absoluteString] UTF8String], &db,
-                           SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE | SQLITE_OPEN_URI, NULL);
-  if (result == SQLITE_OK) {
-    char *errMsg;
-    result = sqlite3_exec(db, [query UTF8String], NULL, NULL, &errMsg);
-    if (result != SQLITE_OK) {
-      MSLogError([MSAppCenter logTag], @"Query \"%@\" failed with error: %d - %@", query, result,
-                 [[NSString alloc] initWithUTF8String:errMsg]);
-    }
-  } else {
-    MSLogError([MSAppCenter logTag], @"Failed to open database for non-selection query with result: %d.", result);
+  return [self executeWithDatabase:^int(void * db) {
+    return [MSDBStorage executeNonSelectionQuery:query inDatabase:db];
+  }];
+}
+
++ (int)executeNonSelectionQuery:(NSString *)query inDatabase:(void *)db {
+  char *errMsg;
+  int result = sqlite3_exec(db, [query UTF8String], NULL, NULL, &errMsg);
+  if (result != SQLITE_OK) {
+    MSLogError([MSAppCenter logTag], @"Query \"%@\" failed with error: %d - %@", query, result,
+               [[NSString alloc] initWithUTF8String:errMsg]);
   }
-  sqlite3_close(db);
-  return SQLITE_OK == result;
+  return result;
 }
 
 - (NSArray<NSArray *> *)executeSelectionQuery:(NSString *)query {
-  NSMutableArray<NSMutableArray *> *entries = [[NSMutableArray<NSMutableArray *> alloc] init];
-  sqlite3 *db = NULL;
+  __block NSArray<NSArray *> *entries = nil;
+  [self executeWithDatabase:^int(void * db) {
+    entries = [MSDBStorage executeSelectionQuery:query inDatabase:db];
+    return SQLITE_OK;
+  }];
+  return entries != nil ? entries : [NSArray<NSArray *> new];
+}
+
++ (NSArray<NSArray *> *)executeSelectionQuery:(NSString *)query inDatabase:(void *)db {
```

Comment:
Please rename `inDatabase` by `inOpenedDatabase`, we want to make it explicit that the database has to be opened before it's passed to this method.